pkg/installmanager/installmanager_test.go

```diff
@@ -647,6 +651,55 @@ func TestInstallManagerSSH(t *testing.T) {
 		})
 	}
 }
+func TestInstallManagerSSHKnownHosts(t *testing.T) {
+	apis.AddToScheme(scheme.Scheme)
+
+	tests := []struct {
+		name         string
+		knownHosts   []string
+		expectedFile string
+	}{
+		{
+			name: "single ssh known host",
+			knownHosts: []string{
+				"192.168.86.100 ecdsa-sha2-nistp256 FOOBAR",
+			},
+			expectedFile: `192.168.86.100 ecdsa-sha2-nistp256 FOOBAR`,
+		},
+		{
+			name: "multiple ssh known hosts",
+			knownHosts: []string{
+				"192.168.86.100 ecdsa-sha2-nistp256 FOOBAR",
+				"192.168.86.101 ecdsa-sha2-nistp256 FOOBAR2",
+				"192.168.86.102 ecdsa-sha2-nistp256 FOOBAR3",
+			},
+			expectedFile: `192.168.86.100 ecdsa-sha2-nistp256 FOOBAR
+192.168.86.101 ecdsa-sha2-nistp256 FOOBAR2
+192.168.86.102 ecdsa-sha2-nistp256 FOOBAR3`,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			tempDir, err := ioutil.TempDir("", "installmanagersshknownhosts")
+			require.NoError(t, err)
+			defer os.RemoveAll(tempDir)
+
+			im := InstallManager{
+				log:      log.WithField("test", test.name),
+				LogLevel: "debug",
```

Comment:
Neither `LogLevel` nor `WorkDir` are used by this test. Why set them? Moreover, if we want to set `WorkDir`, it should be a different directory than the directory passed as an argument to `writeSSHKnownHosts`.